### PR TITLE
Speedy issue 6573

### PR DESCRIPTION
**Pull request by nickchapman-da**
Address issue: #6573

Note to reviewers: 
The (small) substantive change is in the 1st commit. The 2nd commit is just `scalafmt` rearranging everything!

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 1]
```diff
@@ -468,10 +465,12 @@ class Runner(
...
-    source
-      .viaMat(msgFlow)(Keep.right[NotUsed, T])
-      .toMat(getTriggerSink(name, acs, submit))(Keep.both)
-      .run()
+    val (t, futureValue) =
+      source
+        .viaMat(msgFlow)(Keep.right[NotUsed, T])
+        .toMat(getTriggerSink(name, acs, submit))(Keep.both)
+        .run()
+    (t, futureValue.map(SEValue(_)))
```
Comment (cocreature) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:473]: If we always return an `SValue` can we just change the return type? It doesn’t seem useful to wrap it here.

Comment (nickchapman-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:473]: There's lots of callers in the test which would need adapting. Happy to do that if that's what you prefer.

Comment (cocreature) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:473]: Let’s do that. Should hopefully be easy enough to change.

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 2]
```diff
@@ -392,40 +389,40 @@ class Runner(
...
+        val newState = handleStepResult(value, submit)
+        newState
+      }))(Keep.right[NotUsed, Future[SValue]])
   }
 
-  def makeApp(func: AExpr, args: Array[SExprAtomic]): AExpr = {
+  def makeApp(func: AExpr, values: Array[SValue]): AExpr = {
+    val args: Array[SExprAtomic] = values.map(SEValue(_))
+    // We can safely introduce a let-expression here to bind the `func` expression,
+    // because there are no stack-references in `args`, since they are pure speedy values.
```
Comment (hurryabit) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:425]: Given that the coding is a lot cleaner now, I wouldn't have asked for the comment anymore. 😃
But it doesn't do harm either.

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala (hunk 1 above)
Comment (hurryabit) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:473]: Let's do it but in a separate PR.

Comment (nickchapman-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala]: Done. Turned out most callers funnelled through a single point `toResult` where the change could be made.
Thanks.

**Pull request merged**